examples/src/Examples.js

```diff
@@ -83,10 +86,23 @@ export default class Examples extends Component {
     return null;
   }
 
+  processEvents() {
+    if (this.eventQueue.length > 0) {
+      let event = this.eventQueue[0];
+      this.setState({
+        events: this.eventsExist() ? this.state.events.concat([event]) : [event]
+      }, () => {
+        this.eventQueue.shift();
+        this.processEvents();
+      });
+    }
+  }
+
   onEventAdded(event) {
-    this.setState({
-      events: this.eventsExist() ? this.state.events.concat([event]) : [event]
-    });
+    this.eventQueue.push(event);
+    if(this.eventQueue.length === 1) {
```

Comment:
Absolutely. Since every event is added to the queue (on line 102), that means that if the queue length is 1, then the queue was empty. It should **only** call `processEvents` when the queue is empty because if the queue is not empty it means that a `processEvents` loop is already running on the queue (it will run while there are events in the queue left to process) and I don't want to start another one, just add the new event to the queue.

The initial problem was that since unfocusing and focusing happens so fast while moving between suggestions, it ended up only recording the focus event and not the unfocus event. I figured that if you call `setState` before another call to `setState` has completed and it is modifying the same state, only the last call to `setState` will apply, because that is what I was observing. In any case, I implemented the queue to serialize the event adding process and only add new events after all of the others have been successfully added.

I tried to implement `onEventAdded` as least verbose as possible, but I think it affects readability and can be confusing, so I can reimplement to explicitly check for an empty queue.